tensorflow/python/keras/applications/efficientnet_weight_update_util.py

```diff
@@ -0,0 +1,307 @@
+    use_ema: Bool, whether to use ExponentialMovingAverage result or not
+  """
+  model_name_keras = keras_model.name
+  model_name_tf = model_name_keras_to_tf(model_name_keras)
+  keras_model.save_weights(path_h5)
+
+  keras_weights = get_h5_names(path_h5)
+  tf_weights = get_tf_names(path_ckpt)
+  blocks_keras = get_keras_blocks(keras_weights)
+  blocks_tf = get_tf_blocks(tf_weights)
+
+  with tf.compat.v1.Session() as sess:
+    saver = tf.compat.v1.train.import_meta_graph(f'{path_ckpt}.meta')
+    saver.restore(sess, path_ckpt)
+    graph = tf.compat.v1.get_default_graph()
+    v_all = tf.compat.v1.global_variables()
+
+    for keras_block, tf_block in zip(blocks_keras, blocks_tf):
+      print(f'working on block {keras_block}, {tf_block}')
+      for keras_name in keras_weights:
+        if keras_block in keras_name:
+          tf_name = keras_name_to_tf_name_block(keras_name, keras_block=keras_block, tf_block=tf_block, use_ema=use_ema, model_name_tf=model_name_tf)
+          for v in v_all:
+            if v.name == tf_name:
+              v_val = sess.run(v)
+              with h5py.File(path_h5, 'a') as f:
+                v_prev = f[keras_name][()]
+                f[keras_name].write_direct(v_val)
+              print(f'writing from: {tf_name}\n  to: {keras_name}')
+              print(f'  average change: {abs(v_prev - v_val).mean()}')
+              v_all.remove(v)
+              break
+          else:
+            raise ValueError(f'{keras_name} has no match in ckpt file')
+
+    for keras_name in keras_weights:
+      if any([x in keras_name for x in ['stem', 'top', 'predictions', 'probs']]):
+        tf_name = keras_name_to_tf_name_stem_top(keras_name, use_ema=use_ema, model_name_tf=model_name_tf)
+        for v in v_all:
+          if v.name == tf_name:
+            v_val = sess.run(v)
+            with h5py.File(path_h5, 'a') as f:
+              v_prev = f[keras_name][()]
+              try:
+                f[keras_name].write_direct(v_val)
+              except:
+                raise ValueError(f'weight in {tf_name} does not ift into {keras_name}')
+            print(f'writing from: {tf_name}\n  to: {keras_name}')
+            print(f'  average change: {abs(v_prev - v_val).mean()}')
+            v_all.remove(v)
+            break
+
+
+def model_name_keras_to_tf(model_name_keras):
+  """Infer model name in both keras and tf implementations"""
+  model_name_tf = model_name_keras.replace('efficientnet', 'efficientnet-')
+  return model_name_tf
+
+
+def get_h5_names(path_h5):
+  """Get list of variable names from the h5 file """
+  h5_namelst = []
+  def append_to_lst(x):
+    h5_namelst.append(x)
+
+  with h5py.File(path_h5, 'r') as f:
+    for x in f.keys():
+      f[x].visit(append_to_lst)
+
+  # all weights end with ':0'
+  h5_namelst = [x for x in h5_namelst if ':' in x]
+
+  # append group name to the front
+  h5_namelst = ['/'.join([x.split('/')[0], x]) for x in h5_namelst]
+    
+  return h5_namelst
```

Comment:
Just curious, can we get the same value from model.weights, which should also return a list of variable tensor with proper name.

Having to save the model and parse the h5 file seems to be bit complicated.

Reply:
Yes we can. I just somehow did not think about doing this way.

So actually I can read these variables and assign to the model, and save altogether, instead of accessing h5 file so many times.